docs/build.md

```diff
@@ -227,6 +229,47 @@ spec:
       name: icr-knbuild
 ```
 
+### Runtime-Image
+
+Runtime-image is a new image composed with build-strategy outcomes. On which you can compose a multi-stage build action, copying parts out the original image into a new one. This feature allows replacing the base-image of any container-image, creating final leaner images, and other use-cases that require multi-stage build.
+
+The following examples illustrates how to use this feature:
+
+```yml
+apiVersion: build.dev/v1alpha1
+kind: Build
+metadata:
+  name: nodejs-ex-runtime
+spec:
+  strategy:
+    name: buildpacks-v3
+    kind: ClusterBuildStrategy
+  source:
+    url: https://github.com/sclorg/nodejs-ex.git
+  output:
+    image: image-registry.openshift-image-registry.svc:5000/build-examples/nodejs-ex
+  runtime:
+    base:
+      image: docker.io/node:latest
+    workDir: /home/node/app
+    directories:
+      - /workspace/source:/home/node/app
+    entrypoint:
+      - npm
+      - start
+```
+
+This build will produce a NodeJS based application where a single directory is imported from the image built by buildpacks strategy.
+
+Please consider the description of the attributes in `.spec.runtime`:
+
+- `spec.runtime.base`: specifies the runtime base-image to be used, using Image as type
+- `spec.runtime.workDir`: path to WORKDIR in runtime-image
+- `spec.runtime.env`: runtime-image additional environment variables, key-value
+- `spec.runtime.labels`: runtime-image additional labels, key-value
+- `spec.runtime.directories`: list of directories paths to be copied to runtime-image, those can be defined as `<source>:<destination>` split by colon (`:`)
+- `spec.runtime.entrypoint`: entrypoint command, specified as a list
```

Comment:
I think it would be of value to allow specifying a user. Rational is that many containers run as root by default while this is not allowed in all container runtime environments. Probably the generated Dockerfile should then create the user, `chown` all the mapped directories to the user and define the corresponding `USER` command.

Reply:
Should we default to `USER 10001`?